### ISSUE TITLE
Git module recursive option

### ISSUE THREAD

**Issue by luxifer** [ansible/ansible#7927]
##### Issue Type:

Bug Report
##### Ansible Version:

`ansible-playbook 1.6.3`
##### Environment:

N/A
##### Summary:

The `recursive` option of the git module seems not to be used.
##### Steps To Reproduce:
- create a task with git
- set option `recursive=no`

``` yaml
- git: repo=git@github.com:some/repo.git
       dest=/some/path
       recursive=no
```
##### Expected Results:

I want the ansible git module to clone my repo without the submodule (by not passing the `--recursive` to git)
##### Actual Results:

Ansible pass the `--recursive` to git so my repo was clone with the submodules

**Comment by jimi-c** [comment]
I believe we fixed this in https://github.com/ansible/ansible/pull/7033. Can you try with the devel branch and see if you still see the same issue?

**Comment by luxifer** [comment]
Thanks. I'll try it later this day.
Le 26 juin 2014 05:31, "James Cammarata" notifications@github.com a écrit
:

**Comment by cvctimlindsay** [comment]
I'm running 1.6.6 and having same issue. Here's dpkg info on ansible version.

Package: ansible
Status: install ok installed
Priority: optional
Section: admin
Installed-Size: 3142
Maintainer: Henry Graham (hzgraham) Henry.Graham@mail.wvu.edu
Architecture: all
Version: 1.6.6-trusty-unstable-ppa
Depends: python, python-support (>= 0.90), python-jinja2, python-yaml, python-paramiko, python-httplib2, sshpass, python-setuptools
Conffiles:
 /etc/ansible/hosts 7b96f47043374c481808dd8179a09473
 /etc/ansible/ansible.cfg 6b478ec8ab0bc4feffadf3037dbc264c
Description: Ansible Application
 Ansible is a extra-simple tool/API for doing 'parallel remote things' over SSH executing commands, running "modules", or executing larger 'playbooks' that can serve as a configuration management or deployment system.
Homepage: http://ansible.github.com/

**Comment by prune998** [comment]
same issue with 1.6.6 here. 

Task : 

```
  - name: "git pull in {{ build_path }}"
    git:  repo={{ git_url }} dest={{ build_path }}
          recursive=no
          version="release-{{ release_version }}"
          accept_hostkey=True
```

Resultant processes (ps auxwww) : 

```
/usr/bin/git submodule update --init --recursive --remote
```

**Comment by jimi-c** [comment]
@prune998 / @cvctimlindsay as stated above, the issue was fixed in devel, and will be included in 1.7. It is not currently in the 1.6.x branch.

**Comment by mpdehaan** [comment]
Closing as this was fixed on the development branch per James above.   Let me know if you disagree.
